### PR TITLE
feat(pair): Add new state for non-Firefox desktop users at pair/unsupported

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* Users on non-Firefox desktop should be shown a more helpful page

This commit:
* Creates a new state on pair/unsupported for non-Firefox, non-Android and non-iOS users to inform them they need to download Firefox to proceed
* Adds cad_redirect_desktop_view and cad_redirect_desktop_download glean events

closes FXA-10008

---

LMK if I did something wrong regarding the couple of unrelated Glean events. I just ran `yarn glean-generate` after rebasing and that's what it spat out.

Screenshot sent, Eduardo approved:

Chatted with Ross also about 1) the `isDesktopNonFirefox` check, around `!iOS` and `!isAndroid` and 2) the functionality of this link. We could have potentially repurposed the link on `update_firefox` but a new window link-out was preferred.

Also got the alt text from #accessibility.

### CODE REVIEW

File: packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
-    <div class="bg-image-pair-fail"></div>
+    {{#isDesktopNonFirefox}}
+      <div class="bg-no-ff-desktop bg-contain bg-center bg-no-repeat h-44" aria-label="{{#t}}Pink exclamation mark denoting Firefox is missing from a desktop device{{/t}}" role="img"></div>
+      <h2 id="pair-unsupported-header" class="card-header focus:outline-none mt-5" tabindex="-1">{{#t}}Oops! It looks like you’re not using Firefox.{{/t}}</h2>
+      <div class="flex bg-orange-50 px-4 py-3 mt-6 rounded-lg mb-4">
+        <div class="bg-icon-warning bg-contain bg-center bg-no-repeat w-6"></div>
+        <p class="ps-3">{{#t}}Switch to Firefox and open this page to connect another device.{{/t}}</p>
+      </div>
+      <div class="flex">
+        <a id="download-firefox" href="https://www.mozilla.org/firefox/new/" target="_blank" class="cta-primary cta-xl">{{#t}}Download Firefox{{/t}}<span class="sr-only">{{#t}}Opens in new window{{/t}}</span></a>
```
Comment (flodolo) [inline packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache:22]: What is `sr-only` stand for? Is that an accessibility note to explain that the link opens in a new window?

**Review by flodolo — APPROVED** [review]

File: packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache (hunk 1 above)
Comment (LZoog) [reply packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache:22]: Correct, `sr-only` is a class we have [in Tailwind](https://tailwindcss.com/docs/screen-readers) that's visually hidden from users but still read to screenreaders. We mostly use it in our [React LinkExternal component](https://github.com/mozilla/fxa/blob/main/packages/fxa-react/components/LinkExternal/index.tsx#L43), but the page in this PR is still in Backbone and we don't have a component for it there, so I manually added it.

File: packages/fxa-content-server/app/scripts/views/pair/unsupported.js [hunk 2]
```diff
@@ -2,17 +2,41 @@
...
+    'click #download-firefox': 'downloadFirefoxEngage',
+  };
+
+  downloadFirefoxEngage() {
+    GleanMetrics.cadRedirectDesktop.download();
+  }
+
+  setInitialContext(context) {
+    const uap = this.getUserAgent();
+    // Assume the user is on non-Firefox desktop in this case.
```
Comment (chenba) [inline packages/fxa-content-server/app/scripts/views/pair/unsupported.js:25]: I'm not sure where the assumption is here 😅

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml [hunk 3]
```diff
@@ -1501,6 +1503,42 @@ cad_mobile_pair_use_app:
     data_sensitivity:
       - interaction
 
+cad_redirect_desktop:
+  view:
+    type: event
+    description: |
+      User viewed the "Switch to Firefox" page state after trying to access the pair flow
```
Comment (chenba) [inline packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:1510]: While probably technically correct, the inclusion of 'state' makes this a little awkward to read.

**Review by chenba — APPROVED** [review]
LGTM

**Pull request merged**